web/src/engine/websites/MangaTown.ts

```diff
@@ -19,7 +39,7 @@ export default class extends DecoratableMangaScraper {
     }
 
     public override async FetchPages(chapter: Chapter): Promise<Page[]> {
```

Comment:
This method should avoid making dozens of requests for each page (e.g., responsive usability, avoid timeouts).
Consider using the HTML pages itself as page links (e.g., https://www.mangatown.com/manga/goblin_slayer/c090/2.html) and then doing this JS eval magic to get the image link in the `FetchImage` method.

Reply:
> making dozens of requests for each page? 

I fail to see how we are "making dozens of requests for each page".  `FetchPages` does one request per page (to the endpoint) + 1 initial request. FetchImage only fetch the image.

What is true is that FetchPages is making too many requests.

Ill change it so Fetchpages will just return Pages with needed Parameters { chapterId , pageNumber} 
FetchImage will use the endpoint and eval magic to get real picture url and download it.